File: java/src/main/java/sammancoaching/PhysicalPrinter.java
package sammancoaching;

import java.util.HashMap;
import java.util.Map;

enum Toner {
    Black, Cyan, Magenta, Yellow;
}

public class PhysicalPrinter {
    protected boolean isAvailable;
    private final int jobCount;
    protected Map<Toner, Integer> tonerLevels = new HashMap<>();

    public PhysicalPrinter() {
        this.jobCount = 0; // Assuming job count is initialized to zero
    }

    public boolean isAvailable() {
        return isAvailable;
    }

    public int getTonerLevelPercentage(Toner toner) {
        return tonerLevels.get(toner);
    }

    public int getJobCount() {
        return jobCount;
    }

    public boolean startDocument(boolean isSummary, boolean isPdf, String name) {
        throw new UnsupportedOperationException("Can't call this from a unit test");
    }

    public void endDocument() {
        throw new UnsupportedOperationException("Can't call this from a unit test");
    }
}

File: kotlin/src/main/kotlin/sammancoaching/PhysicalPrinter.kt
package sammancoaching

enum class Toner {
    Black, Cyan, Magenta, Yellow
}

open class PhysicalPrinter {
    var isAvailable: Boolean = false
    private val jobCount: Int = 0 // Assuming job count is initialized to zero
    protected val tonerLevels: MutableMap<Toner, Int> = mutableMapOf()


    fun getTonerLevelPercentage(toner: Toner): Int {
        return tonerLevels[toner] ?: 0
    }

    fun getJobCount(): Int {
        return jobCount
    }

    open fun startDocument(isSummary: Boolean, isPdf: Boolean, name: String): Boolean {
        throw UnsupportedOperationException("Can't call this from a unit test")
    }

    open fun endDocument() {
        throw UnsupportedOperationException("Can't call this from a unit test")
    }
}